pkg/kubelet/kubelet.go

```diff
@@ -932,12 +929,20 @@ func (kl *Kubelet) pullImage(img string, ref *api.ObjectReference) error {
 
 // Kill all running containers in a pod (includes the pod infra container).
 func (kl *Kubelet) killPod(runningPod kubecontainer.Pod) error {
+
 	// Send the kills in parallel since they may take a long time.
 	errs := make(chan error, len(runningPod.Containers))
 	wg := sync.WaitGroup{}
 	for _, container := range runningPod.Containers {
 		wg.Add(1)
 		go func(container *kubecontainer.Container) {
+			// Call the networking plugin for teardown.
```

Comment:
@yifan-gu I think I will file another issue for this, there are a ton of places in the code and in issues where this is hinted at, and it's really orthogonal from the thrust of this PR.